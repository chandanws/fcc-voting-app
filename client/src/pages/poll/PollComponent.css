.chart-form-wrapper {
  display: flex;
  flex-direction: column;
}

@media only screen and (min-width: 1000px) {
  .chart-form-wrapper {
    flex-direction: row;
  }

  .chart__form {
    order: -1;
    align-self: center;
  }
}
