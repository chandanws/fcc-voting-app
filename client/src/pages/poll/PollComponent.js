import React, { Component } from "react";
import PropTypes from "prop-types";
import { Typography } from "../../Typography/Typography";
import {
  PieChart,
  Pie,
  Legend,
  Sector,
  Cell,
  Label,
  ResponsiveContainer
} from "recharts";
import { Form, FormElement } from "../../form/FormComponent";
import CommentContainer from "../../comment/CommentContainer";
import "./PollComponent.css";

const COLORS = ["#0088FE", "#00C49F", "#FFBB28", "#FF8042"];

export default class PollComponent extends Component {
  state = {
    value: "something",
    dummyData: [
      { name: "B", value: 400 },
      { name: "A", value: 200 },
      { name: "C", value: 600 }
    ],
    replyValue: ""
  };

  handleChange = e => this.setState({ value: e.target.value });

  handleSubmit = e => {
    e.preventDefault();
    alert(this.state.value);
  };
  render() {
    return (
      <div style={{ width: "100%" }}>
        {/* TODO: Change text to be based on data from redux*/}
        <Typography component="h1" text="Best Actor" />
        <div className="chart-form-wrapper">
          <ResponsiveContainer aspect={1}>
            <PieChart style={{ fontSize: "2rem" }} height={400}>
              <Legend verticalAlign="top" height={36} />
              <Pie
                label
                fill="#8884d8"
                dataKey="value"
                data={this.state.dummyData}
              >
                {this.state.dummyData.map((entry, index) => (
                  <Cell key={index} fill={COLORS[index % COLORS.length]} />
                ))}
              </Pie>
            </PieChart>
          </ResponsiveContainer>
          <Form className="chart__form" onSubmit={this.handleSubmit}>
            <label className="form__label">
              Select option:
              <select
                style={{
                  fontSize: "2rem",
                  marginLeft: "1rem",
                  border: "1px solid black"
                }}
                value={this.state.value}
                onChange={this.handleChange}
              >
                <option value="something">Something</option>
                <option value="user">User</option>
                <option value="weird">Weird</option>
                <option value="anti">Anti</option>
              </select>
            </label>
          </Form>
        </div>
        <Typography component="h2" text="Comments" />
        <form
          className="comment__form"
          onSubmit={e => {
            e.preventDefault();
          }}
        >
          <textarea
            className="comment__textarea"
            onChange={e => this.setState({ replyValue: e.target.value })}
          />
          <button className="comment__submit">Submit</button>
        </form>

        <CommentContainer />
      </div>
    );
  }
}
