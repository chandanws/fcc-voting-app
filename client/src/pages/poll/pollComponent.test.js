import React from "react";
import PollComponent from "./PollComponent";
import { shallow } from "enzyme";

describe("<PollComponent />", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<PollComponent />);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleChange", () => {
    it("should change state based on selected value", () => {
      wrapper.instance().handleChange({ target: { value: "user" } });
      wrapper.update();
      expect(wrapper.state().value).toBe("user");
    });
  });

  describe("handleSubmit", () => {});
});
