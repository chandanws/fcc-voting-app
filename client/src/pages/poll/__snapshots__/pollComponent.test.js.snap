// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PollComponent /> should render correctly 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Typography
    component="h1"
    text="Best Actor"
  />
  <div
    className="chart-form-wrapper"
  >
    <ResponsiveContainer
      aspect={1}
      debounce={0}
      height="100%"
      width="100%"
    >
      <PieChart
        barCategoryGap="10%"
        barGap={4}
        cx="50%"
        cy="50%"
        endAngle={360}
        height={400}
        innerRadius={0}
        layout="centric"
        margin={
          Object {
            "bottom": 5,
            "left": 5,
            "right": 5,
            "top": 5,
          }
        }
        outerRadius="80%"
        reverseStackOrder={false}
        stackOffset="none"
        startAngle={0}
        style={
          Object {
            "fontSize": "2rem",
          }
        }
      >
        <Legend
          align="center"
          height={36}
          iconSize={14}
          layout="horizontal"
          verticalAlign="top"
        />
        <Pie
          animationBegin={400}
          animationDuration={1500}
          animationEasing="ease"
          cx="50%"
          cy="50%"
          data={
            Array [
              Object {
                "name": "B",
                "value": 400,
              },
              Object {
                "name": "A",
                "value": 200,
              },
              Object {
                "name": "C",
                "value": 600,
              },
            ]
          }
          dataKey="value"
          endAngle={360}
          fill="#8884d8"
          hide={false}
          innerRadius={0}
          isAnimationActive={true}
          label={true}
          labelLine={true}
          legendType="rect"
          minAngle={0}
          nameKey="name"
          outerRadius="80%"
          paddingAngle={0}
          startAngle={0}
          stroke="#fff"
        >
          <Cell
            fill="#0088FE"
            key="0"
          />
          <Cell
            fill="#00C49F"
            key="1"
          />
          <Cell
            fill="#FFBB28"
            key="2"
          />
        </Pie>
      </PieChart>
    </ResponsiveContainer>
    <Form
      className="chart__form"
      onSubmit={[Function]}
    >
      <label
        className="form__label"
      >
        Select option:
        <select
          onChange={[Function]}
          style={
            Object {
              "border": "1px solid black",
              "fontSize": "2rem",
              "marginLeft": "1rem",
            }
          }
          value="something"
        >
          <option
            value="something"
          >
            Something
          </option>
          <option
            value="user"
          >
            User
          </option>
          <option
            value="weird"
          >
            Weird
          </option>
          <option
            value="anti"
          >
            Anti
          </option>
        </select>
      </label>
    </Form>
  </div>
  <Typography
    component="h2"
    text="Comments"
  />
  <form
    className="comment__form"
    onSubmit={[Function]}
  >
    <textarea
      className="comment__textarea"
      onChange={[Function]}
    />
    <button
      className="comment__submit"
    >
      Submit
    </button>
  </form>
  <CommentContainer />
</div>
`;
